networkx/algorithms/isomorphism/tests/test_match_helpers.py

```diff
@@ -1,27 +1,55 @@
+from math import isclose
 from operator import eq
 
+import pytest
+
 import networkx as nx
 from networkx.algorithms import isomorphism as iso
 
 
-def test_categorical_node_match():
-    nm = iso.categorical_node_match(["x", "y", "z"], [None] * 3)
+@pytest.mark.parametrize(
+    "nm",
+    [
+        (iso.categorical_node_match(["x", "y", "z"], [None] * 3)),
+        (iso.numerical_node_match(["x", "y", "z"], [None] * 3)),
+    ],
+)
+def test_node_match(nm):
     assert nm(dict(x=1, y=2, z=3), dict(x=1, y=2, z=3))
     assert not nm(dict(x=1, y=2, z=2), dict(x=1, y=2, z=1))
 
 
+@pytest.mark.parametrize(
+    "full_match, flow_match, min_flow_match, id_match",
+    [
+        (
+            iso.generic_multiedge_match(
+                ["id", "flowMin", "flowMax"], [None] * 3, [eq] * 3
+            ),
+            iso.generic_multiedge_match(["flowMin", "flowMax"], [None] * 2, [eq] * 2),
+            iso.generic_multiedge_match("flowMin", None, eq),
+            iso.generic_multiedge_match("id", None, eq),
+        ),
+        (
+            iso.numerical_multiedge_match(["id", "flowMin", "flowMax"], [3] * 3),
+            iso.numerical_multiedge_match(["flowMin", "flowMax"], [4, 5]),
+            iso.numerical_multiedge_match("flowMin", 1.6),
+            iso.numerical_multiedge_match("id", 1.7),
+        ),
+    ],
+)
```

Comment:
Sorry, I may have led you astray here with my previous comment by not being specific enough... what I had envisioned was instead parametrizing the *graph type*, not the checks themselves, i.e. something like:

```python
@pytest.mark.parametrize("graph_type", (nx.MultiGraph, nx.MultiDiGraph))
class TestGenericMultiEdgeMatch:
    def setup_method(self, graph_type):
        self.G1 = graph_type()
        self.G2 = graph_type()
        ... etc. - the remainder of the test could remain unchanged.
```

Reply:
> Sorry, I may have led you astray here with my previous comment by not being specific enough... what I had envisioned was instead parametrizing the _graph type_, not the checks themselves, i.e. something like:
> 
> ```python
> @pytest.mark.parametrize("graph_type", (nx.MultiGraph, nx.MultiDiGraph))
> class TestGenericMultiEdgeMatch:
>     def setup_method(self, graph_type):
>         self.G1 = graph_type()
>         self.G2 = graph_type()
>         ... etc. - the remainder of the test could remain unchanged.
> ```

Here you are looking at the graph type only, but maybe you did not have a look at the difference in multi-edge match methods(ie. numerical and generic). The graph type is not a big deal there, I just preferred using a different type, I could still have left it to be the same.

Reply:
@nsengiyumva-wilberforce sorry for the delay in responding! We still need to test for different graph types too. Could you also test with MultiGraph with `parametrize`.
No worries if you don't have the bandwidth to work on this anymore. Thanks for your contributions!